units/crons.go

```diff
@@ -631,6 +633,65 @@ func PopulateAgentDeployJobs(env evergreen.Environment) amboy.QueueOperation {
 
 }
 
+// PopulateAgentMonitorDeployJobs enqueues the jobs to deploy the agent monitor
+// to any host in which: (1) the agent monitor has not been deployed yet, (2)
+// the agent's last communication time has exceeded the threshold or (3) has
+// already been marked as needing to redeploy a new agent monitor.
+func PopulateAgentMonitorDeployJobs(env evergreen.Environment) amboy.QueueOperation {
+	return func(ctx context.Context, queue amboy.Queue) error {
+		flags, err := evergreen.GetServiceFlags()
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		if flags.AgentStartDisabled {
+			grip.InfoWhen(sometimes.Percent(evergreen.DegradedLoggingPercent), message.Fields{
+				"message": "agent start disabled",
+				"impact":  "agents are not deployed",
+				"mode":    "degraded",
+			})
+			return nil
+		}
+
+		// The agent monitor deploy job will atomically clear the
+		// NeedsNewAgentMonitor field to prevent other jobs from running
+		// concurrently.
+		if err = host.UpdateAll(host.LastCommunicationTimeElapsed(time.Now()), bson.M{"$set": bson.M{
+			host.NeedsNewAgentMonitorKey: true,
```

Comment:
I would avoid doing work like this in the population job.... 

We should talk a bit more about this, but I think we should try and find other methods to ensure exclusion.